internal/git/git.go

```diff
@@ -154,18 +149,148 @@ func (g *GitOperations) GetBranchShas(ctx context.Context, branch string) (Branc
 	err = decoder.Decode(&hydratorFile)
 	if err != nil {
 		return BranchShas{}, fmt.Errorf("could not unmarshal metadata file: %w", err)
 	}
-	shas.Dry = hydratorFile.DrySHA
+	shas.Dry = hydratorFile.DrySha
 	logger.V(4).Info("Got dry branch sha", "branch", branch, "sha", shas.Dry)
 
 	return shas, nil
 }
 
+func (g *GitOperations) GetShaMetadataFromFile(ctx context.Context, sha string) (v1alpha1.CommitShaState, error) {
+	logger := log.FromContext(ctx)
+
+	if gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext) == "" {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("no repo path found for repo %q", g.gitRepo.Name)
+	}
+
+	metadataFileStdout, stderr, err := g.runCmd(ctx, gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext), "show", sha+":hydrator.metadata")
+	if err != nil {
+		logger.Error(err, "could not git show file", "gitError", stderr)
+		return v1alpha1.CommitShaState{}, nil
+	}
+	logger.V(4).Info("Got metadata file", "sha", sha, "file", metadataFileStdout)
+
+	var hydratorFile v1alpha1.HydratorMetadata
+	err = json.Unmarshal([]byte(metadataFileStdout), &hydratorFile)
+	if err != nil {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("could not unmarshal metadata file: %w", err)
+	}
+
+	hydratorMetadataDate := v1.Time{}
+	if hydratorFile.Date != "" {
+		hDate, err := time.Parse(time.RFC3339, hydratorFile.Date)
+		if err != nil {
+			return v1alpha1.CommitShaState{}, fmt.Errorf("failed to parse date %q: %w", hydratorFile.Date, err)
+		}
+		hydratorMetadataDate = v1.Time{Time: hDate}
+	}
+
+	commitState := v1alpha1.CommitShaState{
+		Sha:        hydratorFile.DrySha,
+		CommitTime: hydratorMetadataDate,
+		RepoURL:    hydratorFile.RepoURL,
+		Commands:   hydratorFile.Commands,
+		Author:     hydratorFile.Author,
+		Subject:    hydratorFile.Subject,
+		Body:       hydratorFile.Body,
+		References: hydratorFile.References,
+	}
+
+	return commitState, nil
+}
+
+func (g *GitOperations) GetShaMetadataFromGit(ctx context.Context, sha string) (v1alpha1.CommitShaState, error) {
+	// logger := log.FromContext(ctx)
+
+	if gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo)+g.pathContext) == "" {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("no repo path found for repo %q", g.gitRepo.Name)
+	}
+
+	commitTime, err := g.GetShaTime(ctx, sha)
+	if err != nil {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("failed to get commit time for hydrated SHA %q: %w", sha, err)
+	}
+
+	commitAuthor, err := g.GetShaAuthor(ctx, sha)
+	if err != nil {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("failed to get commit author for hydrated SHA %q: %w", sha, err)
+	}
+
+	commitSubject, err := g.GetShaSubject(ctx, sha)
+	if err != nil {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("failed to get commit time for hydrated SHA %q: %w", sha, err)
+	}
+
+	commitBody, err := g.GetShaBody(ctx, sha)
+	if err != nil {
+		return v1alpha1.CommitShaState{}, fmt.Errorf("failed to get commit body for hydrated SHA %q: %w", sha, err)
+	}
```

Comment:
If number of execs became a concern, we could probably shove this into one newline-delimited `git show` command. Just something to consider.